cql3/statements/truncate_statement.cc

```diff
@@ -92,12 +92,15 @@ truncate_statement::execute(query_processor& qp, service::query_state& state, co
     if (qp.db().find_schema(keyspace(), column_family())->is_view()) {
         throw exceptions::invalid_request_exception("Cannot TRUNCATE materialized view directly; must truncate base table instead");
     }
-    auto timeout_in_ms = std::chrono::duration_cast<std::chrono::milliseconds>(get_timeout(state.get_client_state(), options));
-    return qp.proxy().truncate_blocking(keyspace(), column_family(), timeout_in_ms).handle_exception([](auto ep) {
-        throw exceptions::truncate_exception(ep);
-    }).then([] {
-        return ::shared_ptr<cql_transport::messages::result_message>{};
-    });
+
```

Comment:
The parlance for this is "coroutinize".

Reply:
> It also adds unit tests for truncate table with tablets.

This should be a separate commit. The changes are unrelated.

Reply:
I've separated the commit and changed the commit message to correct parlance.